NPE when calling put("/api/edit/" + fileId)

As of ad86024023 in develop (and probably sooner, since that was just a doc change) we're seeing a 500 error and NPE when calling api.EditDDI.createNewDraftVersion(

Caused by: java.lang.NullPointerException
	at org.glassfish.grizzly.http.server.Request.getRemoteAddr(Request.java:1219)
	at org.apache.catalina.connector.Request.getRemoteAddr(Request.java:1453)
	at org.apache.catalina.connector.RequestFacade.getRemoteAddr(RequestFacade.java:531)
	at javax.servlet.ServletRequestWrapper.getRemoteAddr(ServletRequestWrapper.java:242)
	at javax.servlet.ServletRequestWrapper.getRemoteAddr(ServletRequestWrapper.java:242)
	at edu.harvard.iq.dataverse.engine.command.DataverseRequest.<init>(DataverseRequest.java:127)
	at edu.harvard.iq.dataverse.api.EditDDI.createNewDraftVersion(EditDDI.java:196)
	at edu.harvard.iq.dataverse.api.EditDDI.edit(EditDDI.java:140)

- develop: https://jenkins.dataverse.org/job/IQSS-dataverse-develop/1167/consoleFull
- a PR: https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-8953/3/testReport/edu.harvard.iq.dataverse.api/EditDDIIT/testUpdateVariableMetadata/

The test is failing here:

```
EditDDIIT.testUpdateVariableMetadata(EditDDIIT.java:153)
```

Here's what the test is trying to do:

        Response editDDIResponseNewDraft = UtilIT.editDDI(stringOrigXml, origFileId, apiToken);
        assertEquals(200, editDDIResponseNewDraft.getStatusCode());

--- a/src/main/java/edu/harvard/iq/dataverse/api/EditDDI.java
+++ b/src/main/java/edu/harvard/iq/dataverse/api/EditDDI.java
@@ -35,7 +35,6 @@ import javax.ejb.Stateless;
 import javax.inject.Inject;
 import javax.persistence.EntityManager;
 import javax.persistence.PersistenceContext;
-import javax.servlet.http.HttpServletRequest;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.Context;
 import javax.ws.rs.Path;
@@ -91,9 +90,6 @@ public class EditDDI  extends AbstractApiBean {
     DataverseSession session;
 
     private List<FileMetadata> filesToBeDeleted = new ArrayList<>();
-
-    @Context
-    protected HttpServletRequest httpRequest;
 
     private VariableMetadataUtil variableMetadataUtil;
 
@@ -193,7 +189,7 @@ public class EditDDI  extends AbstractApiBean {
         Command<Dataset> cmd;
         try {
 
-            DataverseRequest dr = new DataverseRequest(apiTokenUser, httpRequest);
+            DataverseRequest dr = createDataverseRequest(apiTokenUser);
             cmd = new UpdateDatasetVersionCommand(dataset, dr, fm);
             ((UpdateDatasetVersionCommand) cmd).setValidateLenient(true);
             dataset = commandEngine.submit(cmd);
@@ -335,7 +331,7 @@ public class EditDDI  extends AbstractApiBean {
         }
         Command<Dataset> cmd;
         try {
-            DataverseRequest dr = new DataverseRequest(apiTokenUser, httpRequest);
+            DataverseRequest dr = createDataverseRequest(apiTokenUser);
             cmd = new UpdateDatasetVersionCommand(dataset, dr);
             ((UpdateDatasetVersionCommand) cmd).setValidateLenient(true);
             commandEngine.submit(cmd);

